tests/model_serving/model_server/kueue/conftest.py

```diff
@@ -0,0 +1,207 @@
+from typing import Generator, Any, Dict
+
+import pytest
+from kubernetes.dynamic import DynamicClient
+from _pytest.fixtures import FixtureRequest
+from utilities.kueue_utils import (
+    create_local_queue,
+    create_cluster_queue,
+    create_resource_flavor,
+    LocalQueue,
+    ClusterQueue,
+    ResourceFlavor,
+)
+from ocp_resources.namespace import Namespace
+from utilities.constants import ModelAndFormat, KServeDeploymentType
+from utilities.inference_utils import create_isvc
+from utilities.serving_runtime import ServingRuntimeFromTemplate
+from ocp_resources.secret import Secret
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.serving_runtime import ServingRuntime
+from utilities.constants import RuntimeTemplates, ModelFormat
+import logging
+
+BASIC_LOGGER = logging.getLogger(name="basic")
+
+
+def kueue_resource_groups(
+    flavor_name: str,
+    cpu_quota: int,
+    memory_quota: str,
+) -> list[Dict[str, Any]]:
+    return [
+        {
+            "coveredResources": ["cpu", "memory"],
+            "flavors": [
+                {
+                    "name": flavor_name,
+                    "resources": [
+                        {"name": "cpu", "nominalQuota": cpu_quota},
+                        {"name": "memory", "nominalQuota": memory_quota},
+                    ],
+                }
+            ],
+        }
+    ]
+
+
+@pytest.fixture(scope="class")
+def kueue_cluster_queue_from_template(
+    request: FixtureRequest,
+    admin_client: DynamicClient,
+) -> Generator[ClusterQueue, Any, Any]:
+    if request.param.get("name") is None:
+        raise ValueError("name is required")
+    with create_cluster_queue(
+        name=request.param.get("name"),
+        client=admin_client,
+        resource_groups=kueue_resource_groups(
+            request.param.get("resource_flavor_name"), request.param.get("cpu_quota"), request.param.get("memory_quota")
+        ),
+        namespace_selector=request.param.get("namespace_selector", {}),
```

Comment:
I'm wondering if we want to scope it down to our test namespace to not affect potentially other tests running in other namespaces?